lib/membrane_rtsp/server/logic.ex

```diff
@@ -0,0 +1,229 @@
+                  session_state: :init
+                ]
+
+    @type t :: %__MODULE__{
+            socket: :inet.socket(),
+            rtp_socket: :inet.socket() | nil,
+            rtcp_socket: :inet.socket() | nil,
+            request_handler: module(),
+            request_handler_state: term(),
+            configured_media: Server.Handler.configured_media_context(),
+            session_id: binary(),
+            session_state: :init | :ready | :playing | :paused,
+            session_timeout: non_neg_integer()
+          }
+  end
+
+  @spec allowed_methods() :: [binary()]
+  def allowed_methods(), do: @allowed_methods
+
+  @spec process_request(binary(), State.t()) :: State.t()
+  def process_request(raw_request, %State{} = state) do
+    {response, state} =
+      case Request.parse(raw_request) do
+        {:ok, request} ->
+          {response, state} = do_handle_request(request, state)
+          {maybe_add_cseq_header(response, request), state}
+
+        {:error, _reason} ->
+          {Response.new(400), state}
+      end
+
+    response
+    |> Response.with_header("Server", @server)
+    |> Response.stringify()
+    |> then(&mockable(:gen_tcp).send(state.socket, &1))
+
+    state
+  end
+
+  defp do_handle_request(%Request{method: method}, state) when method not in @allowed_methods do
+    {Response.new(501), state}
+  end
+
+  defp do_handle_request(%Request{method: "OPTIONS"}, state) do
+    Response.new(200)
+    |> Response.with_header("Public", Enum.join(@allowed_methods, ", "))
+    |> then(&{&1, state})
+  end
+
+  defp do_handle_request(%Request{method: "GET_PARAMETER"}, state) do
+    {Response.new(200), state}
+  end
+
+  defp do_handle_request(%Request{method: "DESCRIBE"} = request, state) do
+    {response, request_handler_state} =
+      state.request_handler.handle_describe(request, state.request_handler_state)
+
+    {response, %{state | request_handler_state: request_handler_state}}
+  end
+
+  defp do_handle_request(%Request{method: "SETUP"} = request, state)
+       when state.session_state in [:init, :ready] do
+    with {:ok, transport_opts} <- Request.parse_transport_header(request),
+         :ok <- validate_transport_parameters(transport_opts, state) do
+      {response, request_handler_state} =
+        state.request_handler.handle_setup(request, state.request_handler_state)
+
+      {response, state} = do_handle_setup_response(request, response, transport_opts, state)
+      {response, %{state | request_handler_state: request_handler_state}}
+    else
+      _error ->
+        {Response.new(400), %{state | request_handler_state: state.request_handler_state}}
+    end
+  end
+
+  defp do_handle_request(%Request{method: "PLAY"}, state)
+       when state.session_state in [:ready, :paused] do
+    {response, request_handler_state} =
+      state.request_handler.handle_play(state.configured_media, state.request_handler_state)
+
+    response = response |> Response.with_header("Session", state.session_id)
```

Comment:
We should add the timeout to the session header whenever we send it

`Session  = "Session" ":" session-id [ ";" "timeout" "=" delta-seconds ]`